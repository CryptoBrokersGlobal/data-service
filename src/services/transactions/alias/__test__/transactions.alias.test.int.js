const createService = require('..');
const { createPgDriver } = require('../../../../db');

// both broken aliases, have 3 txs each on this ID
const TX_ID = '21BmkGpk98wWHrCydjwTFvHWPfQxtH5NEQd9X7wUpi5R';
const TX_ID_2 = 'HbgSfoQ7ftAF3TUQNovbFnohsRK7AdztFznrJHTqH7F7';

const loadConfig = require('../../../../loadConfig');
const options = loadConfig();

const drivers = {
  pg: createPgDriver(options),
};

const service = createService({
  drivers,
  emitEvent: () => () => null,
});

describe('Alias transaction service', () => {
  describe('get', () => {
    it('fetches real tx', async done => {
      service
        .get(TX_ID)
        .run()
        .promise()
        .then(x => {
          expect(x).toMatchSnapshot();
          done();
        })
        .catch(e => done(JSON.stringify(e)));
    });
    it('returns null for unreal tx', async () => {
      const tx = await service
        .get('UNREAL')
        .run()
        .promise();

      expect(tx).toBe(null);
    });
  });

  describe('mget', () => {
    it('fetches real txs with nulls for unreal', async done => {
      service
        .mget([TX_ID, 'UNREAL', TX_ID_2])
        .run()
        .promise()
        .then(xs => {
          expect(xs).toMatchSnapshot();
          done();
        })
        .catch(e => done(JSON.stringify(e)));
    });
  });
});
