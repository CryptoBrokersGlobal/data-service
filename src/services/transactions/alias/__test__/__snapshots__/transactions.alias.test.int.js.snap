// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alias transaction service get fetches real tx 1`] = `
Object {
  "__type": "transaction",
  "data": Object {
    "alias": "wykop.pl",
    "fee": "0.001",
    "height": 922106,
    "id": "21BmkGpk98wWHrCydjwTFvHWPfQxtH5NEQd9X7wUpi5R",
    "sender": "3P2DQSKgE7Hw9VhHfu5TkJxsh5GFRfShQpK",
    "senderPublicKey": "FTY2T2HLvyR7XQTvn5Eu56B8qGXcvosutTRq5TatcTMp",
    "signature": "54qNaDCCC3LoprhNcd6C4ytRbB2chDpYxEusmyVgmPg7PNQjvi7Ckc13mBn1M3zf8T5YgzfWDMnvE9rRLYyGSkSu",
    "timestamp": 2018-03-17T20:22:43.000Z,
    "type": 10,
    "version": 1,
  },
}
`;

exports[`Alias transaction service mget fetches real txs with nulls for unreal 1`] = `
Object {
  "__type": "list",
  "data": Array [
    Object {
      "__type": "transaction",
      "data": Object {
        "alias": "wykop.pl",
        "fee": "0.001",
        "height": 922106,
        "id": "21BmkGpk98wWHrCydjwTFvHWPfQxtH5NEQd9X7wUpi5R",
        "sender": "3P2DQSKgE7Hw9VhHfu5TkJxsh5GFRfShQpK",
        "senderPublicKey": "FTY2T2HLvyR7XQTvn5Eu56B8qGXcvosutTRq5TatcTMp",
        "signature": "54qNaDCCC3LoprhNcd6C4ytRbB2chDpYxEusmyVgmPg7PNQjvi7Ckc13mBn1M3zf8T5YgzfWDMnvE9rRLYyGSkSu",
        "timestamp": 2018-03-17T20:22:43.000Z,
        "type": 10,
        "version": 1,
      },
    },
    null,
    Object {
      "__type": "transaction",
      "data": Object {
        "alias": "wykop",
        "fee": "0.001",
        "height": 922106,
        "id": "HbgSfoQ7ftAF3TUQNovbFnohsRK7AdztFznrJHTqH7F7",
        "sender": "3P2DQSKgE7Hw9VhHfu5TkJxsh5GFRfShQpK",
        "senderPublicKey": "FTY2T2HLvyR7XQTvn5Eu56B8qGXcvosutTRq5TatcTMp",
        "signature": "4oA22v2NsNNJafLJVEqwSJF74XvN1pxh9d72k2pe8kheMjrTf8E4exLUGAoGKKrZYcw9UzxVgFoeNNBQjcqRajHW",
        "timestamp": 2018-03-17T20:22:39.000Z,
        "type": 10,
        "version": 1,
      },
    },
  ],
}
`;
